clients/go/commands/createInstance_command.go

```diff
@@ -121,6 +140,19 @@ func (cmd *CreateInstanceCommand) BPMNProcessId(id string) CreateInstanceCommand
 	return cmd
 }
 
+func (cmd *CreateInstanceCommand) WithResult() CreateInstanceWithResultCommandStep1 {
+	return &CreateInstanceWithResultCommand{
+		SerializerMixin: cmd.SerializerMixin,
+		request: &pb.CreateWorkflowInstanceWithResultRequest{
+			Request:        cmd.request,
+			RequestTimeout: int64(cmd.requestTimeout / time.Millisecond),
+		},
+		gateway:        cmd.gateway,
+		requestTimeout: cmd.requestTimeout,
```

Comment:
This requestTimeout is used as the grpc deadline. So I think it would be better to give it a value slight greater than cmd.requestTimeout which would be used as the timeout in the gateway. Otherwise, by the time gateway times out the requests grpc channel will be closed. We did something similar for long polling timeout.

Reply:
Sorry I missed that part of the implementation. I also added a offset to the grpc deadline now, see https://github.com/zeebe-io/zeebe/blob/bb1bf47b7a718b1fb292dcea7fe3ed34c8b1e60b/clients/go/commands/createInstance_command.go#L170-L171